packages/common/src/server.ts

```diff
@@ -505,11 +489,41 @@ export class YogaServer<
 
       return result
     } catch (error: unknown) {
-      return getErrorResponse({
-        status: 500,
-        errors: [new Error((error as Error)?.message ?? 'Unexpected Error.')],
-        fetchAPI: this.fetchAPI,
-      })
+      const finalResponseInit = {
+        status: 200,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+
+      const errors = handleError(error)
+      for (const error of errors) {
+        if (error.extensions?.http) {
+          if (
+            error.extensions.http.status &&
+            error.extensions?.http.status > finalResponseInit.status
+          ) {
+            finalResponseInit.status = error.extensions.http.status
+          }
+          if (error.extensions.http.headers) {
+            Object.assign(
+              finalResponseInit.headers,
+              error.extensions.http.headers,
+            )
+          }
+          // Remove http extensions from the final response
+          delete error.extensions.http
```

Comment:
```suggestion
          error.extensions.http = undefined
```

I rather prefer this instead of delete

Reply:
I'm afraid it keeps the property so current users might have extensions with an empty http field.

Reply:
`undefined` is always omitted when being serialized.

Reply:
Oh you're right actually :D undefined is not available in JSON. Cool.